src/pyhf/workspace.py

```diff
@@ -161,3 +161,186 @@ def data(self, model, with_aux=True):
         if with_aux:
             observed_data += model.config.auxdata
         return observed_data
+
+    def _prune_and_rename(
```

Comment:
could this maybe more generally be a visitor-like pattern in which prune returns `None`, and rename returns the renamed object?

Reply:
If you're thinking of traversal, I tried.. but if you have a better way to implement it... The `_prune_and_rename` isn't a public api though. If you prefer, i can just copy/paste that code into the two methods and get rid of this one.